### PR TITLE
Keep hidden conversations invisible

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

Address https://github.com/microsoft/chat-copilot/issues/371

### Description

1. When a selected conversation is hidden, keep it hidden and show a message.
2. After deleting a conversation, and all other conversations are hidden, automatically create a new chat.

image

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by glahaye — DISMISSED** [review]

**Review by alliscode — APPROVED** [review]

File: webapp/src/components/chat/ChatRoom.tsx [hunk 1]
```diff
@@ -105,6 +106,22 @@ export const ChatRoom: React.FC = () => {
...
 
+    if (conversations[selectedId].hidden) {
+        return (
+            <div className={classes.root}>
+                <div className={classes.scroll}>
+                    <div className={classes.history}>
+                        <h3>
+                            This conversation is not visible in the app because{' '}
+                            {Features[FeatureKeys.MultiUserChat].label} is disabled. Please enable the feature in the
+                            settings to view the conversation, or select a different or create a new conversation.
```
Comment (dehoward) [inline webapp/src/components/chat/ChatRoom.tsx:117]:
```suggestion
                            settings to view the conversation, select a different one, or create a new conversation.
```

File: webapp/src/libs/hooks/useChat.ts [hunk 2]
```diff
@@ -371,8 +371,11 @@ export const useChat = () => {
             .then(() => {
                 dispatch(deleteConversation(chatId));
 
-                // If there is only one chat left, create a new chat
-                if (Object.keys(conversations).length <= 1) {
+                if (Object.values(conversations).filter((c) => c.id != chatId).length === 0) {
+                    // If there is no chat left, create a new chat
+                    void createChat();
+                } else if (Object.values(conversations).filter((c) => !c.hidden && c.id !== chatId).length === 0) {
```
Comment (dehoward) [inline webapp/src/libs/hooks/useChat.ts:377]: we can remove this first condition since its handled by the second

**Review by dehoward — CHANGES_REQUESTED** [review]

**Review by crickman — DISMISSED** [review]

File: webapp/src/components/chat/ChatRoom.tsx (hunk 1 above)
Comment (dehoward) [reply webapp/src/components/chat/ChatRoom.tsx]: this wasn't updated in your commit

Comment (TaoChenOSU) [reply webapp/src/components/chat/ChatRoom.tsx]: It's updated. You might have to select the latest diff.

**Review by dehoward — APPROVED** [review]

**Pull request merged**